docs/tracing.md

```diff
@@ -0,0 +1,74 @@
+# Tracing
+
+## Introduction
+
+Instrumentation based tracing is implemented with
+[`log`](https://github.com/rust-lang/log) and
+[`log_instrument`](https://github.com/JonathanWoollett-Light/log-instrument),
+outputting a `Trace` level log when entering and exiting every function.
+
+It is disabled by default at compile-time, as such the
+**tracing functionality has no impact on the release binary.**
+
+To enable tracing, remove `"max_level_debug"` from the features
+for `log` under [`../src/vmm/Cargo.toml`](../src/vmm/Cargo.toml):
+
+```diff
+- log = { version = "0.4.17", features = ["std", "serde", "max_level_debug"] }
++ log = { version = "0.4.17", features = ["std", "serde"] }
+```
+
+This will result in a massive increase in the binary size and a
+massive regression in performance. To mitigate the performance affects
+you can use filtering.
```

Comment:
nit:
```suggestion
This will result in a significant increase in binary size and
regression in performance. To mitigate the performance impact
you can filter the tracing output as described in the next section.
```

Reply:
I was switching between `massive` and `significant`, I liked `massive` since it makes people less likely to ask `how significant` and simply informs them its too big to be used in production. I think `significant` can be used with some ballpark percentages.

Reply:
I like the added information on how large the impacts are :)